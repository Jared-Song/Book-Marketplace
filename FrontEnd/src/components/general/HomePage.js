import HorizontalMenu from "./HorizontalMenu";
import React from "react";
import Grid from "@material-ui/core/Grid";
import BookCard from "./BookCard"
import BigMenu from "./BigMenu";
const menuItems = [
  {
    title: "Home",
    selected: true,
  },
  {
    title: "New Releases",
    selected: false,
  },
  {
    title: "Comming Soon",
    selected: false,
  },
  {
    title: "Best Sellers",
    selected: false,
  },
  {
    title: "View All Books",
    selected: false,
  },
];
const categories = [
  {
    title: "Romance",
    selected: false,
  },
  {
    title: "Literature & Fiction",
    selected: false,
  },
  {
    title: "Textbooks & Study Guides",
    selected: false,
  },
  {
    title: "Mystery, Thriller & Suspense",
    selected: false,
  },
  {
    title: "Science Fiction & Fantasy",
    selected: false,
  },
  {
    title: "Childrenâ€™s Books",
    selected: false,
  },
  {
    title: "Family & Lifestyle",
    selected: false,
  },
  {
    title: "Teen & Young Adult",
    selected: false,
  },
  {
    title: "Humour & Entertainment",
    selected: false,
  },
  {
    title: "Health, Fitness & Nutrition",
    selected: false,
  },
  {
    title: "Religion, philosophy & Social Sciences",
    selected: false,
  },
  {
    title: "Biographies & Memoies",
    selected: false,
  },
];

export default function HomePage() {
  return (
    <Grid container>
      <Grid item xs={2}>
      <BigMenu/>
      </Grid>
      <Grid item xs={10}>
      </Grid>

    </Grid>
  );
}
