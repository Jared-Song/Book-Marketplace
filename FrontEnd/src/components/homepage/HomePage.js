import React from "react";
import Grid from "@material-ui/core/Grid";
import BigMenu from "../general/BigMenu";
import BookListCard from "../general/BookListCard";

import useAxios from "axios-hooks";
import router from "next/router";

export default function HomePage() {
  const [selectedMenu, setSelectedMenu] = React.useState("Home");
  const [newReleaseNumber, setNewReleaseNumber] = React.useState(4);
  const [bestSellersNumber, setBestSellersNumber] = React.useState(4);
  const [randomNumber, setRandomNumber] = React.useState(4);
  React.useEffect(() => {
    if (selectedMenu === "New Releases") {
      setNewReleaseNumber(20);
    } else if (selectedMenu === "Best Sellers") {
      setBestSellersNumber(20);
    } else if (selectedMenu === "Maybe You Like") {
      setRandomNumber(20);
    } else {
      setNewReleaseNumber(4);
      setBestSellersNumber(4);
      setRandomNumber(4);
    }
  }, [selectedMenu]);

  const [{ data: book1, loading: loading1, error: error1 }, refetch1] =
    useAxios(
      process.env.NEXT_PUBLIC_BROWSE_URL + `newReleases/${newReleaseNumber}`
    );

  const [{ data: book2, loading: loading2, error: error2 }, refetch2] =
    useAxios(
      process.env.NEXT_PUBLIC_BROWSE_URL + `bestSellers/${bestSellersNumber}`
    );

  const [{ data: book3, loading: loading3, error: error3 }, refetch3] =
    useAxios(process.env.NEXT_PUBLIC_BROWSE_URL + `random/${randomNumber}`);

  if (loading1 && error1) {
    return <SimpleLoadingPlaceholder />;
  }
  if (loading2 && error2) {
    return <SimpleLoadingPlaceholder />;
  }

  if (loading3 && error3) {
    return <SimpleLoadingPlaceholder />;
  }
  return (
    <Grid container>
      <Grid item xs={2}>
        <BigMenu
          selectedMenu={selectedMenu}
          setSelectedMenu={setSelectedMenu}
        />
      </Grid>
      <Grid item xs={10}>
        {book1 && ["Home", "New Releases"].includes(selectedMenu) && (
          <BookListCard
            books={book1}
            title="New Releases"
            handleClick={
              selectedMenu !== "New Releases"
                ? () => {
                    setNewReleaseNumber((old) => {
                      return old + 4;
                    });
                  }
                : undefined
            }
          />
        )}
        {book2 && ["Home", "Best Sellers"].includes(selectedMenu) && (
          <BookListCard
            books={book2}
            title="Best Sellers"
            handleClick={
              selectedMenu !== "Best Sellers"
                ? () => {
                    setBestSellersNumber((old) => {
                      return old + 4;
                    });
                  }
                : undefined
            }
          />
        )}
        {book3 && ["Home", "Maybe You Like"].includes(selectedMenu) && (
          <BookListCard
            books={book3}
            title="Maybe You Like"
            handleClick={
              selectedMenu !== "Maybe You Like"
                ? () => {
                    setRandomNumber((old) => {
                      return old + 4;
                    });
                  }
                : undefined
            }
          />
        )}
      </Grid>
    </Grid>
  );
}
