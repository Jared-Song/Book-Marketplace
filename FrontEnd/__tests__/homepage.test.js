import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import HomePage from "../src/components/homepage/HomePage";


jest.mock("next/router", () => ({
  useRouter() {
    return {
      route: "/",
      pathname: "",
      query: "",
      asPath: "",
    };
  },
}));


it("renders correctly", () => {
  const tree = renderer.create(<HomePage />).toJSON();
  expect(tree).toMatchSnapshot();
});

