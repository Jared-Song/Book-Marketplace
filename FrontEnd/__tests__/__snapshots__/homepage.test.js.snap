// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-flex-start"
    >
      <div
        className="makeStyles-root-1"
      >
        <nav
          className="MuiList-root MuiList-padding"
        >
          <div
            aria-disabled={true}
            className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters Mui-disabled MuiListItem-button Mui-selected Mui-disabled"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={-1}
          >
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                Home
              </span>
            </div>
          </div>
          <div
            aria-disabled={false}
            className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                New Releases
              </span>
            </div>
          </div>
          <div
            aria-disabled={false}
            className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                Best Sellers
              </span>
            </div>
          </div>
          <div
            aria-disabled={false}
            className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                Maybe You Like
              </span>
            </div>
          </div>
        </nav>
        <hr
          className="MuiDivider-root"
        />
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <span
          className="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-wave"
          style={
            Object {
              "height": 220,
              "width": undefined,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
